### DISCUSSION TITLE
`wisdom2eg`  테스트 데이터셋 리뷰하기

### DISCUSSION THREAD

**Discussion by eubinean** [wisdomify/wisdomify#78] [Q&A]
## Why?
여러 문장이 이어진 데이터를 테스트셋으로 사용하는 것이, 정말 옳은 방향일까? 우리 모델의 타겟 사용자는 속담을 배우고자하는  학생이다 (e.g. 초등학생, 외국인) - 그들이 입력으로 넣을 문장이 이렇게 기사와 비슷한 문장이라고 가정할수는 없다.
그렇다면, 제대로된 테스트 셋을 만들기 위해선, 모든  문장을 한 문장으로, 요약을 해서 (혹은 핵심이 되는 문장 하나만 추출) 해서 구축을 하는 편이 낫지 않을까?
e.g. - 핑계없는 무덤의 용례;
```
이혼을 선택한 사피영에게 그는 말한다. 뭐가 옳았는지는 나중에 돼봐야 알고요. 참아서 잘 된 경우도 있고 안 참아서 잘 된 경우도 있고요. 의지와 약속을 지키지 않는 남자들이 이해가 가지 않는다며 친구분들은 어떠냐는 질문에 서반은 3분의1은 이혼했다며 그 이유를 이렇게 답한다. 성격차이도 있고 양쪽 말 들어봐야겠지만 상대방이 못돼서 갈라설 수밖에 없었다는 친구도 있고 다양해요.
```
를 다음과 같이 변경:
```
성격차이도 있고 양쪽 말 들어봐야겠지만 상대방이 못돼서 갈라설 수밖에 없었다는 친구도 있고 다양해요.
```
그러고 보니, 풀고자하는 문제를 좀 더 정확하게  정의할 필요가 있다.  이를테면 그냥 "용례 -> 속담"말고, "한문장 용례 -> 속담"식으로. 그래야 모델링이 편해진다.

**Comment by ArtemisDicoTiar** [comment]
이 데이터는 순전히 모델의 성능 테스트용이기 때문에 context가 clear한 경우(이 경우 문장이 길어지거나 복수 문장을 사용할 가능성이 높), unclear한 경우 (문장이 짧거나, 불필요한 정보가 과하게 적혀있을 것이다) 모두 사용 해볼 필요가 있지 않나 싶다.

물론 대부분의 사용자가 짧게 문장을 input할 가능성이 있지만, 긴 문장 + clear context로 좋은 성능이 나온다는 걸 입증한다면 사용자가 사용하는 단계에서 "쿼리를 최대한 자세하게 적어서 검색하세요"라는 메시지를 띄울 수 있을 것이다.

가령 아무리 tip of the tongue 상황이더라도 "산 넘어 산"라는 속담을 묘사하라고 했을 때

1. "아 그... **약속을 자주 안 지키는 친구**가 뭔 잘못을 하더라도 계속 **변명**하는 상황에 알맞는 속담이 뭐가 있지?"라는 식으로 쿼리를 작성하게 되면
context가 불충분하기 때문에 "핑계없는 무덤"? "산 넘어 산"? 이 두개의 속담이 헷갈릴수도 그리고 올바르지 않는 답을 보여줄 수도 있다.

2. "약속을 잘 안 지키는 친구가 약속시간에 늦을 때마다 이런저런 변명을 들이밀며 이래서 늦을 수 밖에 없었다며 자신은 잘못이 없다고 말하는 데 이때 적합한 속담이 뭐지?"
이렇게 context가 구체적이면 유저가 원하는 정답을 금방 얻을 수도 있을 것.

교육이 목적이기 때문에 최대한 accurate해야하며, 유저로 하여금 최대한 자세한 묘사, 용례를 생각해내면서 해당 속담/관용구의 사용에 대해 좀더 고민하게 하여 언어 학습을 유도할수도 있을 것.

**Comment by ArtemisDicoTiar** [comment]
+ 생각보다 유저에게 원하는 바를 clear하고 accurate하지만 짧게 말해달라는 건 어려울 수 있다. 
대체로 뭔가를 묘사하거나 관련 예제를 생각해낼 때, <1문장... (부족한데?) 2문장... (흐음...) 3문장... (아 길다...) 1.5문장... (흠... 더 줄일 수 있나? 아닌가 말이 이상해지나?) 1 or 2문장> 과 같은 flow를 보일 것 같다. (~내가 그렇게 한다는 건 비밀~)
오히려 줄이면서 의미 변질, 시간 소요가 발생할 것이다.

결론: 긴 문장 하나든, 짧은 문장 두세개든 전체적인 길이가 지나치게 긴 것만 아니면 괜찮은 테스트 데이터셋이 아닌가 싶다.
